--- libproxy-0.2.3/configure.ac	2008-06-05 22:56:16.000000000 +0200
+++ libproxy-0.2.3.new/configure.ac	2008-12-18 17:45:19.000000000 +0100
@@ -19,8 +19,9 @@ PKG_CHECK_MODULES(webkit, webkit-1.0, ha
 PKG_CHECK_MODULES(mozjs, xulrunner-js, have_mozjs=yes, 
 		[PKG_CHECK_MODULES(mozjs, firefox-js, have_mozjs=yes,
 		[PKG_CHECK_MODULES(mozjs, mozilla-js, have_mozjs=yes, have_mozjs=no)])])
-PKG_CHECK_MODULES(NetworkManager, NetworkManager, 
-					have_networkmanager=yes, have_networkmanager=no)
+PKG_CHECK_MODULES(NetworkManager, NetworkManager,
+		[PKG_CHECK_MODULES(DBUS, dbus-1, have_networkmanager=yes, have_networkmanager=no)],
+		have_networkmanager=no)
 
 # Environmental Variable
 AC_ARG_WITH([envvar],
@@ -136,12 +137,12 @@ AC_ARG_WITH([networkmanager],
             [test x$have_networkmanager == xyes && with_networkmanager=yes])
 if test x$with_networkmanager = xyes; then
 	if test x$have_networkmanager == xyes; then
-		NETWORKMANAGER_CFLAGS="$NetworkManager_CFLAGS"
-		NETWORKMANAGER_LIBS="$NetworkManager_LIBS"
+		NETWORKMANAGER_CFLAGS="$NetworkManager_CFLAGS $DBUS_CFLAGS"
+		NETWORKMANAGER_LIBS="$NetworkManager_LIBS $DBUS_LIBS"
 		AC_SUBST(NETWORKMANAGER_CFLAGS)
 		AC_SUBST(NETWORKMANAGER_LIBS)
 	else
-		echo "NetworkManager plugin requires: NetworkManager!"
+		echo "NetworkManager plugin requires: NetworkManager and DBus!"
 		exit 1
 	fi
 else
