diff -ruNp libproxy-0.3.1.orig/src/lib/Makefile.am libproxy-0.3.1/src/lib/Makefile.am
--- libproxy-0.3.1.orig/src/lib/Makefile.am	2009-09-29 21:52:50.000000000 +0200
+++ libproxy-0.3.1/src/lib/Makefile.am	2009-12-12 01:27:11.000000000 +0100
@@ -11,7 +11,7 @@ libproxy_la_SOURCES = \
 						url.c url.h
 
 libproxy_la_CFLAGS  = -DMODULEDIR=\"@MODULEDIR@@DIR_SEPARATOR@\" -DSYSCONFDIR=\"@SYSCONFDIR@@DIR_SEPARATOR@\" -std=c99 -D_POSIX_C_SOURCE=200112L
-libproxy_la_LDFLAGS = -lpthread -lm -no-undefined
+libproxy_la_LDFLAGS = -lpthread -lm -ldl -no-undefined
 
 include_HEADERS     = proxy.h
 
diff -ruNp libproxy-0.3.1.orig/src/lib/Makefile.in libproxy-0.3.1/src/lib/Makefile.in
--- libproxy-0.3.1.orig/src/lib/Makefile.in	2009-10-08 21:23:57.000000000 +0200
+++ libproxy-0.3.1/src/lib/Makefile.in	2009-12-12 01:27:29.000000000 +0100
@@ -285,7 +285,7 @@ libproxy_la_SOURCES = \
 						url.c url.h
 
 libproxy_la_CFLAGS = -DMODULEDIR=\"@MODULEDIR@@DIR_SEPARATOR@\" -DSYSCONFDIR=\"@SYSCONFDIR@@DIR_SEPARATOR@\" -std=c99 -D_POSIX_C_SOURCE=200112L
-libproxy_la_LDFLAGS = -lpthread -lm -no-undefined
+libproxy_la_LDFLAGS = -lpthread -lm -ldl -no-undefined
 include_HEADERS = proxy.h
 all: all-am
 
